src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -170,16 +171,57 @@ def _get_storage_sync_endpoint(cloud_name):
         'AzureCloud': 'afs.azure.net',
         'AzureUSGovernment': 'afs.azure.us',
     }
-    return storage_sync_endpoint_mapper.get(cloud_name, 'afs.azure.net')
+    return storage_sync_endpoint_mapper.get(cloud_name, None)
+
+
+def _get_database_server_endpoint(db_prefix, sql_server_hostname, cloud_name):
+    if cloud_name == 'AzureCloud':
+        return db_prefix + '.database.azure.com'
+    if not sql_server_hostname:
```

Comment:
why not do this in `AzureCloud` ?

Reply:
For `AzureCloud`, we know `sql_server_hostname` exists and use hard-coded value as a patch since the value for `AzureCloud sql_server_hostname` returned from ARM metadata is wrong.